sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java

```diff
@@ -242,6 +243,168 @@ public void recordAddressResolutionEnd(String identifier, String exceptionMessag
         }
     }
 
+    private void mergeContactedReplicas(List<URI> otherContactedReplicas) {
+        if (otherContactedReplicas == null) {
+            return;
+        }
+
+        if (this.contactedReplicas == null || this.contactedReplicas.isEmpty()) {
+            this.contactedReplicas = otherContactedReplicas;
+            return;
+        }
+
+        LinkedHashSet<URI> totalContactedReplicas = new LinkedHashSet<>();
```

Comment:
why are we using a `LinkedHashSet`? We can use ArrayList, that should also maintain the ordering if it is required here.

Reply:
The Set<T> is used only when both instances to-be-merged have contacted regions. Set<T> is needed to deduplicate the values - I used LinkedHashSet to still preserve the order in which regions got contacted. otherContactedReplicas is the diagnostics for the first request being merged in here.